tests/unit/Runner/TestSuiteSorterTest.php

```diff
@@ -621,4 +690,23 @@ public function testSuiteSorterDefectsOptions(int $order, bool $resolveDependenc
 
         $this->assertSame($expected, $sorter->getExecutionOrder());
     }
+
+    #[DataProvider('defectsSorterWithDataProviderProvider')]
+    public function testSuiteSorterDefectsWithDataProviderTest(int $order, bool $resolveDependencies, array $runState, array $expected): void
```

Comment:
this newly added test shows that `TestSuiteSorter` works correctly with `DataProviderTestSuite` - it worked correct even before the fix.

I still think how to best test the actual problem this PR fixed in `ResultCacheHandler`